conf/modules.config

```diff
@@ -252,34 +300,73 @@ process {
 
 // UMI Subworkflow
 
+    withName: 'FASTQTOBAM'{
+        ext.prefix       = {"${meta.id}"}
+        ext.args         = { "--read-structures $params.umi_read_structure" }
+        publishDir       = [
+            //specify to avoid publishing, overwritten otherwise
+            enabled: false
+        ]
+    }
+
     withName: 'BAM2FASTQ' {
         ext.args         = '-T RX'
         ext.when         = { params.umi_read_structure }
-    }
-
-    withName: 'CALLUMICONSENSUS' {
-        ext.args         = '-M 1 -S Coordinate'
-        ext.prefix       = {"${meta.id}_umi-consensus"}
+        publishDir       = [
+            //specify to avoid publishing, overwritten otherwise
+            enabled: false
+        ]
     }
 
     if (params.umi_read_structure) {
         withName: "NFCORE_SAREK:SAREK:CREATE_UMI_CONSENSUS:MAPPING_UMI:BWAMEM.*_MEM" {
             ext.args         = { "-K 100000000 -p -C -Y -R ${meta.read_group}" }
             ext.args2        = '-bS'
             ext.prefix       = {"${meta.id}.umi_unsorted"}
+            publishDir       = [
+                //specify to avoid publishing, overwritten otherwise
+                enabled: false
+            ]
         }
 
         withName: ".*DRAGMAP_ALIGN" {
             ext.args2        = '-bS'
             ext.prefix       = {"${meta.id}.umi_unsorted"}
+            publishDir       = [
+                //specify to avoid publishing, overwritten otherwise
+                enabled: false
+            ]
         }
     }
 
     withName: 'SAMBLASTER' {
         ext.args         = '-M --addMateTags'
         ext.prefix       = {"${meta.id}_unsorted_tagged"}
+        publishDir       = [
+            //specify to avoid publishing, overwritten otherwise
+            enabled: false
+        ]
+    }
+    withName: 'GROUPREADSBYUMI' {
+        publishDir       = [
+            [   path: { "${params.outdir}/reports/umi/" },
+                mode: params.publish_dir_mode,
+                pattern: "*.{txt}"
+            ]
+        ]
+    }
+
+    withName: 'CALLUMICONSENSUS' {
+        ext.args         = '-M 1 -S Coordinate'
+        ext.prefix       = {"${meta.id}_umi-consensus"}
```

Comment:
mix of underscores and dashes in this filename - you def want this?

Reply:
I have no issue with mixing `_` and `-` for me `_` would really separate 2 parts of the filename, here the id and the detail of what's in the file; the `-` just joining together 2 parts in that detail.